lib/fs_utils/asset.js

```diff
@@ -1,119 +1,91 @@
 'use strict';
 const debug = require('debug')('brunch:asset');
 const sysPath = require('universal-path');
-const copyFile = require('quickly-copy-file');
-const prettify = require('../utils/helpers').prettify;
-const isIgnored = require('./common').isIgnored;
-const pipeline = require('./pipeline');
-const logger = require('loggy');
+const readFromCache = require('fcache').readFile;
+const processAsset = require('./pipeline').processAsset;
 const BrunchError = require('../utils/error');
+const isIgnored = require('./is_ignored');
 
-// Asset: Simple abstraction on top of static assets that are not compiled.
-
-// Get first parent directory that matches asset convention.
-// getAssetDirectory('app/assets/thing/thing2.html', /assets/)
-// => 'app/assets/'
-// Returns String.
-const getAssetDirectory = (path, convention) => {
-  const split = path.split('/');
-
-  // Creates list similar to:
-  // ['app/', 'app/assets/', 'app/assets/thing/', 'app/assets/thing/item.html/']
-  return split.map((part, index) => {
-    return split.slice(0, index).concat(part, '').join(sysPath.sep);
-  }).find(convention);
+// Returns all parent directories:
+// < parentDirs('app/assets/thing/index.html')
+// > ['app/', 'app/assets/', 'app/assets/thing/']
+const parentDirs = path => {
+  return path.split('/').map((part, index, parts) => {
+    return parts.slice(0, index).concat(part, '').join('/');
+  }).slice(0, -1);
 };
```

Comment:
Hey Artem, thanks for the review.

Indeed, we can do that, however I prefer `{}` and `return` even for single-expression arrows that do not "fit" on one line, because it takes less effort to modify then.

Consider the diff for adding statement before return:

```diff
-  return path.split('/').map((part, index, parts) => (
+  return path.split('/').map((part, index, parts) => {
+    if (index === 42) throw new RangeError();
-    parts.slice(0, index).concat(part, '').join('/')
+    return parts.slice(0, index).concat(part, '').join('/');
-  )).slice(0, -1);
+  }).slice(0, -1);
```

versus just

```diff
  return path.split('/').map((part, index, parts) => {
+   if (index === 42) throw new RangeError();
    return parts.slice(0, index).concat(part, '').join('/');
  }).slice(0, -1);
```